include/oneapi/dpl/pstl/hetero/dpcpp/sycl_traits.h

```diff
@@ -71,10 +71,13 @@ class __set_value;
 template <typename _Comp, typename _Proj>
 struct __predicate;
 
+template <typename _F, typename _Proj>
```

Comment:
Generally, when we add these to sycl_traits, we also add positive and negative tests to check them in `device_copyable.pass`. It may be a bit overkill, but that has been the convention so far.

Reply:
I wont insist, but we normally check both the positive condition and negative condition. (it is device copyable when the relevant fields are device copyable, AND it is not device copyable when at least one relevant template argument is not device copyable)